synapse/cores/common.py

```diff
@@ -62,59 +62,116 @@ class Cortex(EventBus, DataModel, Runtime, s_ingest.IngestApi):
-        self.on('node:del', self._onDelAuthUserRole, form='syn:auth:userrole')
-        self.on('node:del', self._onDelSynTag, form='syn:tag')
-        self.on('node:form', self._onFormSynTag, form='syn:tag')
-
-        self.on('fifo:ack', self._onFifoAck)
-        self.on('fifo:sub', self._onFifoSub)
-
-        # a cortex may have a ref to an axon
-        self.axon = None
-        self.seedctors = {}
+        # Store the link tufo
+        self._link = link
 
+        # Core modules and model related structs
         self.modules = [(ctor, modconf) for ctor, smod, modconf in s_modules.ctorlist]
+        self.coremods = {}  # name:module ( CoreModule() )
         self.modsdone = False
+        self.seedctors = {}
+        self.modelrevlist = []
 
-        self.noauto = {'syn:form', 'syn:type', 'syn:prop'}
+        # Auth helpers
+        self._auth_perms = {}
+        self._auth_roles = s_cache.Cache(onmiss=self._onAuthRolesMiss)
+        self._auth_users = s_cache.Cache(onmiss=self._onAuthUsersMiss)
+        self._auth_rules = s_cache.Cache(onmiss=self._onAuthRulesMiss)
 
-        self.onConfOptSet('modules', self._onSetMods)
-        self.onConfOptSet('caching', self._onSetCaching)
-        self.onConfOptSet('axon:url', self._onSetAxonUrl)
+        # Tag Caches
+        self._core_tags = s_cache.FixedCache(maxsize=10000, onmiss=self._getFormFunc('syn:tag'))
+        self._core_tagforms = s_cache.FixedCache(maxsize=10000, onmiss=self._getFormFunc('syn:tagform'))
 
-        logger.debug('Setting Cortex conf opts')
-        self.setConfOpts(conf)
+        # BusRef for handling named Fifo objects
+        self._core_fifos = s_eventbus.BusRef(ctor=self._initCoreFifo)
 
-        self._link = link
+        # we keep an in-ram set of "ephemeral" nodes which are runtime-only ( non-persistent )
+        self.runt_forms = set()
+        self.runt_props = collections.defaultdict(list)
 
-        self.lock = threading.Lock()
-        self.inclock = threading.Lock()
+        # We keep a cache of recently accessed nodes to redeuce hitting storage layers
+        self.cache_fifo = collections.deque()               # [ ((prop,valu,limt), {
+        self.cache_bykey = {}                               # (prop,valu,limt):( (prop,valu,limt), {iden:tufo,...} )
+        self.cache_byiden = s_cache.RefDict()
+        self.cache_byprop = collections.defaultdict(dict)   # (<prop>,<valu>):[ ((prop, valu, limt),  answ), ... ]
 
-        self.coremods = {}  # name:module ( CoreModule() )
+        # Function helpers
         self.statfuncs = {}
+        self.tufosbymeths = {}
 
-        self.auth = None
-        self.snaps = s_cache.Cache(maxtime=60)
+        # Runtime stats
+        self.formed = collections.defaultdict(int)      # count tufos formed since startup
 
-        self._core_triggers = s_trigger.Triggers()
+        # Misc
+        self.axon = None  # A cortex may have a reference to an Axon
+        self.isok = True
```

Comment:
Do we need `isok`? I'm ok with leaving it there, but it seems pointless right now as nothing is going to change it.

Reply:
I think we can keep it since we may want to plumb it later